core/object.h

```diff
@@ -722,7 +722,9 @@ class Object {
 
 	virtual void get_argument_options(const StringName &p_function, int p_idx, List<String> *r_options) const;
 
-	StringName tr(const StringName &p_message) const; // translate message (internationalization)
+	StringName tr(const StringName &p_message, const StringName &p_context = "") const; // translate message (internationalization)
+	////I'm returning as String here because when I test the API, if I return StringName, I need to wrap it with String() to use format string, which is inconvenient.
```

Comment:
It's a bit weird to have this inconsistency in the API. I guess if we need to return `String` here, we should likely do the same with `tr()`.

I'm not fully up-to-date on how the String and StringName usage will impact GDScript and other languages, maybe @vnen could shed some light.

But it's true that a priori, it makes sense to me for those to return a plain String (StringName is great for comparison / usage as key, but here it will always end up converted to a String for display or formatting I guess).